src/internal/connector/exchange/api/contacts.go

```diff
@@ -55,6 +55,41 @@ func (c Contacts) CreateContactFolder(
 	return mdl, nil
 }
 
+// GetContactFolderByDisplayName fetches contact folder which has the displayName of
+// folderName. This is only done at root level.
+func (c Contacts) GetContactFolderByDisplayName(
+	ctx context.Context,
+	user, folderName string,
+) (models.ContactFolderable, error) {
+	filter := fmt.Sprintf("displayName eq '%s'", folderName)
+	options := &users.ItemContactFoldersRequestBuilderGetRequestConfiguration{
+		QueryParameters: &users.ItemContactFoldersRequestBuilderGetQueryParameters{
+			Filter: &filter,
+		},
+	}
+
+	resp, err := c.Stable.Client().UsersById(user).ContactFolders().Get(ctx, options)
+	if err != nil {
+		return nil, graph.Wrap(ctx, err, "getting contact folder").
+			With("folder name", folderName)
+	}
+
+	// We only expect one folder with provided folderName. Return an error if
+	// multiple folders exist(unlikely) or if no folder was found
+	if len(resp.GetValue()) != 1 {
+		return nil, graph.Wrap(ctx, err, "unexpected number of folders").
+			With("folder count", len(resp.GetValue())).
+			With("folder name", folderName)
```

Comment:
more descriptive keys tend to be more helpful.
```suggestion
			With("returned_folder_count", len(resp.GetValue()))
```

Reply:
✍️